docs/project/os-support.md

```diff
@@ -0,0 +1,38 @@
+# OS Support Matrix
+
+.NET is a cross-platform product, requiring good coverage across [supported OSes](https://github.com/dotnet/core/blob/main/release-notes/10.0/supported-os.md).
+
+This document describes our coverage intent. It is a higher-level description than our pipelines.
+
+Pipelines:
+
+- [Runtime](https://github.com/dotnet/runtime/blob/main/eng/pipelines/coreclr/templates/helix-queues-setup.yml)
+- [Libraries](https://github.com/dotnet/runtime/blob/main/eng/pipelines/libraries/helix-queues-setup.yml)
+
+## Run types
+
+We rely on multiple levels of testing to provide good coverage at reasonable cost. Our testing uses OSes both as a vehicle to test .NET itself and to validate discover distro-specific breakage. This is why we test multiple OSes for each run type.
+
+- Inner loop -- Baseline set of tests that validate correct functional behavior, for PRs and branch builds.

```

Comment:
Maybe worth calling out that inner-loop for main means lots and lots of test coverage. For release branches it's just servicing fixes.

FWIW, it may be simpler to remove this distinction bewteen inner-loop only/outer-loop included tiers. Main will run everything eventually, just not on PR. Release branches could be changed to do the same thing.

Reply:
We need to explains:

- What these different "loops" are for / why they exist to enable others to make reasonable future changes to their pipeline definitions. It was not at all clear to me.
- Why they have different OSes in them and what informs that choice
- Any difference between main and release

We do not need to use tiers to do that. That was the most obvious approach.